src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -110,8 +112,10 @@ public final class StringConcatFactory {
      * While the maximum number of argument slots that indy call can handle is 253,
      * we do not use all those slots, to let the strategies with MethodHandle
      * combinators to use some arguments.
+     *
+     * @since 20
      */
-    private static final int MAX_INDY_CONCAT_ARG_SLOTS = 200;
+    public static final int MAX_INDY_CONCAT_ARG_SLOTS = 200;
```

Comment:
I do not think it's a good idea to make that constant available for everybody given that it's an artefact of the implementation.

Reply:
There have been several requests to make it public in the past.  You really can't use the methods in this class unless you know the value. Better to have the value exposed instead of developers transcribing the value into their code.

Reply:
But it's an implementation details, BTW i wonder if the limitation is still valid, i know that John has changed the implementation of the BSM in that area.

Reply:
Anyway, i think you are right, this can be public

Reply:
If this is a public int field, it will be inlined to class file byte code at compile time, which may not be what you want if this value needs to change in the future. If this needs to be exposed, should it be as a public method?

Reply:
It can also be initialised in a static initialiser, which will make it no longer inlined at use‑sites:
```suggestion
    public static final int MAX_INDY_CONCAT_ARG_SLOTS;
    static { MAX_INDY_CONCAT_ARG_SLOTS = 200; }
```